apps/antalmanac/src/lib/download.ts

```diff
@@ -255,16 +255,30 @@ export function getRRule(bydays: string[], quarter: string) {
 
 export function getEventsFromCourses(
     events = AppStore.getEventsWithFinalsInCalendar(),
-    term = getDefaultTerm().shortName
+    term?: string
 ): EventAttributes[] {
+    /**
+     * The term to assign custom events to.
+     */
+    let customEventTerm = term ?? '';
+
+    // If a designated term has not been determined, assume term of first course event found.
+    if (!customEventTerm) {
+        for (const event of events) {
+            if (!event.isCustomEvent && event.term) {
+                customEventTerm = event.term as typeof customEventTerm;
+            }
+        }
+    }
```

Comment:
We could also augment the `getDefaultTerm` function to accept an optionally provided list of events, and abstract this calculation there. Thoughts? @KevinWu098

Reply:
That makes sense to me -- I think abstracting/extracting it outside of `getEventsFromCourses` is better